sfepy/discrete/common/mappings.py

```diff
@@ -4,6 +4,41 @@
 import numpy as nm
 
 from sfepy.base.base import Struct
+from sfepy.discrete.common.extmods.cmapping import CMapping
+
+
+class PyCMapping(Struct):
+    """
+    Class for storing mapping data. Primary data in numpy arrays.
+    Data for C functions translated to FMFields and embedded in CMapping.
+    """
+    def __init__(self, bf, det, volume, bfg, normal, dim):
+        self.bf = bf
+        self.det = det
+        self.volume = volume
+        self.bfg = bfg
+        self.normal = normal
+
+        self.cmap = CMapping(bf, det, volume, bfg, normal, dim)
```

Comment:
So PyCMapping instance is called geo, and we have geo.cmap.geo. Food for thought... maybe something clearer might be devised. The lowest level cmap.geo is dictated by the Cython/C terms code. What about the upper geo? :)